hydra-cluster/test/Test/Hydra/Cluster/FaucetSpec.hs

```diff
@@ -30,27 +30,29 @@ spec = do
               -- 10 unique outputs
               length (fold utxos) `shouldBe` 10
   describe "returnFundsToFaucet" $
-    it "seedFromFaucet and returnFundsToFaucet should work together" $ do
-      showLogsOnFailure "FaucetSpec" $ \tracer ->
-        withTempDir "hydra-cluster" $ \tmpDir ->
-          withCardanoNodeDevnet (contramap FromCardanoNode tracer) tmpDir $ \node@RunningNode{networkId, nodeSocket} -> do
-            let faucetTracer = contramap FromFaucet tracer
-            actor <- generate $ elements [Alice, Bob, Carol]
-            (vk, _) <- keysFor actor
-            (faucetVk, _) <- keysFor Faucet
-            initialFaucetFunds <- queryUTxOFor networkId nodeSocket QueryTip faucetVk
-            seeded <- seedFromFaucet node vk 100_000_000 faucetTracer
-            returnFundsToFaucet faucetTracer node actor
-            remaining <- queryUTxOFor networkId nodeSocket QueryTip vk
-            finalFaucetFunds <- queryUTxOFor networkId nodeSocket QueryTip faucetVk
-            foldMap txOutValue remaining `shouldBe` mempty
+    prop "seedFromFaucet and returnFundsToFaucet should work together" $
+      withMaxSuccess 10 $
+        forAll (Coin <$> choose (1000000, 10000000000)) $ \coin -> do
+          showLogsOnFailure "FaucetSpec" $ \tracer ->
+            withTempDir "hydra-cluster" $ \tmpDir ->
+              withCardanoNodeDevnet (contramap FromCardanoNode tracer) tmpDir $ \node@RunningNode{networkId, nodeSocket} -> do
+                let faucetTracer = contramap FromFaucet tracer
+                actor <- generate $ elements [Alice, Bob, Carol]
```

Comment:
Should: run generator using `forAll` instead of `generate`

This is now a property and we can run it more directly. This means we need to move this up before `showLogsOnFailure` (where we enter `IO`)